sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java

```diff
@@ -151,7 +152,7 @@ public String toString() {
     }
 
     private static Tag tag(long id) {
-        return Tag.of(TAG_NAME, Strings.padStart(Long.toHexString(id).toUpperCase(), 4, '0'));
+        return Tag.of(TAG_NAME, Strings.padStart(Long.toHexString(id).toUpperCase(Locale.ROOT), 4, '0'));
```

Comment:
@David-Noble-at-work  please review the RNTBD changes.

Reply:
LGTM